src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
I'm surprised. None of the constructors are zero copy. Shouldn't the user be able to have a allocated buffer on the heap and send it via a pointer to another thread/process without doing a memcpy?